x-pack/plugins/maps/public/classes/sources/wms_source/wms_source.tsx

```diff
@@ -5,40 +5,53 @@
  * 2.0.
  */
 
-import { AbstractSource } from '../source';
+import { ReactElement } from 'react';
 import { i18n } from '@kbn/i18n';
+import { RasterTileSource } from 'maplibre-gl';
+import { AbstractSource } from '../source';
 import { getDataSourceLabel, getUrlLabel } from '../../../../common/i18n_getters';
+// @ts-ignore
 import { WmsClient } from './wms_client';
 import { SOURCE_TYPES } from '../../../../common/constants';
 import { registerSource } from '../source_registry';
-
+import { IRasterSource, RasterTileSourceData } from '../raster_source';
+import { WMSSourceDescriptor } from '../../../../common/descriptor_types';
 export const sourceTitle = i18n.translate('xpack.maps.source.wmsTitle', {
   defaultMessage: 'Web Map Service',
 });
 
-export class WMSSource extends AbstractSource {
+export class WMSSource extends AbstractSource implements IRasterSource {
   static type = SOURCE_TYPES.WMS;
-
-  static createDescriptor({ serviceUrl, layers, styles }) {
+  readonly _descriptor: WMSSourceDescriptor;
+  static createDescriptor({ serviceUrl, layers, styles }: Partial<WMSSourceDescriptor>) {
```

Comment:
I wonder if we should make a new type like `WMSSourceConfig` for these arguments similar to how we do with [`XYZTMSSource`](https://github.com/elastic/kibana/blob/main/x-pack/plugins/maps/public/classes/sources/xyz_tms_source/xyz_tms_source.ts#L33)? Using a partial means all the arguments are optional where they should be required since we aren't adding default values in the static method.